Clarify non-deterministic "sort" behavior in absence of an explicit field-list (#2902)
brimdata/zed: docs/language/operators/README.md

| **Description**           | Sort records based on the order of values in the specified named field(s).|
| **Syntax**                | `sort [-r] [-nulls first\|last] [field-list]`                             |
| **Required<br>arguments** | None                                                                      |

> | **Optional<br>arguments** | `[-r]`<br>If specified, results will be sorted in reverse order.<br><br>`[-nulls first\|last]`<br>Specifies where null values (i.e., values that are unset or that are not present at all in an incoming record) should be placed in the output.<br><br>`[field-list]`<br>One or more comma-separated field names by which to sort. Results will be sorted based on the values of the first field named in the list, then based on values in the second field named in the list, and so on.<br><br>If no field list is provided, sort will automatically pick a field by which to sort. The pick is done by examining the first record `sort` receives as input and finding the first field in left-to-right order that is of one of the integer Zed [data types](../data-types/README.md) (`int8`, `uint8`, `int16`, `uint16`, `int32`, `uint32`, `int64`, `uint64`) and if no integer fields are found, the first field of a floating point type (`float16`, `float32`, `float64`) is used. If no fields of these numeric types are found, sorting will be performed on the first field found in left-to-right order that is _not_ of the `time` data type. Note that there are some cases (such as the output of a [grouped aggregation](https://github.com/brimdata/zed/tree/main/docs/language/grouping#note-undefined-order) performed on heterogeneous data) where the first input record to `sort` may vary even when the same query is executed repeatedly against the same data. If you require a query to show deterministic output on repeated execution, an explicit field list should always be provided. |

Reviewer: ```suggestion
| **Optional<br>arguments** | `[-r]`<br>If specified, results will be sorted in reverse order.<br><br>`[-nulls first\|last]`<br>Specifies where null values (i.e., values that are unset or that are not present at all in an incoming record) should be placed in the output.<br><br>`[field-list]`<br>One or more comma-separated field names by which to sort. Results will be sorted based on the values of the first field named in the list, then based on values in the second field named in the list, and so on.<br><br>If no field list is provided, `sort` will automatically pick a field by which to sort. It does so by examining the first input record and finding the first field in left-to-right order that is of a Zed integer [data type](../data-types/README.md) (`int8`, `uint8`, `int16`, `uint16`, `int32`, `uint32`, `int64`, `uint64`) or, if no integer field is found, the first field that is of a floating point data type (`float16`, `float32`, `float64`). If no such numeric field is found, `sort` finds the first field in left-to-right order that is _not_ of the `time` data type. Note that there are some cases (such as the output of a [grouped aggregation](https://github.com/brimdata/zed/tree/main/docs/language/grouping#note-undefined-order) performed on heterogeneous data) where the first input record to `sort` may vary even when the same query is executed repeatedly against the same data. If you require a query to show deterministic output on repeated execution, an explicit field list must be provided. |
```
